src/vs/workbench/contrib/chat/browser/media/chatSessions.css

```diff
@@ -173,6 +173,35 @@
 	animation: codicon-spin 1.5s steps(30) infinite;
 }
 
+/* Override global file icon hiding for chat session icons */
+.monaco-workbench:not(.file-icons-enabled) .chat-sessions-tree-container .predefined-file-icon.codicon-loading::before,
+.monaco-workbench:not(.file-icons-enabled) .chat-sessions-tree-container .predefined-file-icon.codicon-pass::before,
+.monaco-workbench:not(.file-icons-enabled) .chat-sessions-tree-container .predefined-file-icon.codicon-error::before,
+.monaco-workbench:not(.file-icons-enabled) .chat-sessions-tree-container .predefined-file-icon.codicon-circle-outline::before,
+.monaco-workbench:not(.file-icons-enabled) .chat-sessions-tree-container .predefined-file-icon.codicon-chat-sparkle::before {
+	font-family: codicon;
+}
+
+.monaco-workbench:not(.file-icons-enabled) .chat-sessions-tree-container .predefined-file-icon.codicon-loading::before {
+	content: var(--vscode-icon-loading-content) !important;
+}
+
+.monaco-workbench:not(.file-icons-enabled) .chat-sessions-tree-container .predefined-file-icon.codicon-pass::before {
+	content: var(--vscode-icon-pass-content) !important;
```

Comment:
Hardcoding the list of supported icons seems fragile. I'd actually prefer your original approach over this, unless there is some way to re-enable the icons using css in a more generic way

Reply:
Switched back to earlier custom icon implementation.